### ISSUE TITLE
Published OSBA Docker Image Incompatible with Cloud Foundry  

### ISSUE THREAD

**Issue by jeremyrickard** [Azure/open-service-broker-azure#249]
We can currently run OSBA on CF using the go build pack, but that requires the user to clone the repository. CF supports running applications as containers as well, but Cloud Foundry has a requirement of Docker images that we don't currently meet. 

> The Docker image must contain an /etc/passwd file with an entry for the root user. In addition, the home directory and the shell for that root user must be present in the image filesystem.

Should we modify our Dockerfile to include the shell and /etc/passwd?

**Comment by zhongyi-zhang** [comment]
It is common for CF users to clone the repo then push the app. For PCF users, they usually install app by a tile. We can build a tile to release in Pivotal Network after resolving https://github.com/Azure/open-service-broker-azure/issues/129.
So I think there is no strong requirement to make the image work for CF. Still, it is nice to make it work.

**Comment by jeremyrickard** [comment]
Yeah, I thought this was a nice to have. I have a blog post ready to go on using it with pcf dev and thought I'd try to deploy with docker and noticed this one.

We are planning on resolving 129 (we'll support both env variables), so that will be good to get a new tile.

**Comment by krancour** [comment]
@jeremyrickard it sounds like conforming would require us to stop being `FROM scratch`. Is that right?

That's something I'd rather avoid, if possible, because my feeling is that distributions bring a whole lot of new attack vectors along with them. For instance-- right now it's (nearly?) impossible to even exec into a running OSBA container. I'm not keen to undo that.

**Comment by jangaraj** [comment]
You don't need any distribution. Just shell, which starts app commands. I've made my own Golang shell implementation (simplified https://github.com/tianon/gosu with -c flag implementation), which is compiled as a static binary `/bin/sh` and CF is happy with that.

**Comment by krancour** [comment]
@jangaraj including a shell still brings along new attack vectors. As I mentioned, you currently cannot exec into a running OSBA container and I regard that as a feature, tbh.

**Comment by krancour** [comment]
> Any idea how can you exploit it with this fake shell?

This looks pretty safe to me. I don't think I'd understood how narrow in scope the shell in question was.

I think we could consider this.
